pkg/api/types.go

```diff
@@ -1309,3 +1317,27 @@ type ResourceQuotaList struct {
 	// Items is a list of ResourceQuota objects
 	Items []ResourceQuota `json:"items"`
 }
+
+// Secret holds secret data of a certain type
+type Secret struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	Data map[string][]byte `json:"data,omitempty"`
```

Comment:
@thockin @bgrant0607, I was thinking about the `Data` field this weekend -
I think there's a case to be made for using a list of a type like:

```go
type SecretCell struct {
    Name  string
    Value string
    Binary bool
```

I don't think you should have to base64 encode everything - it makes for
kind of a crummy experience IMO, especially when the serialized form is a
string anyway.  The binary bit would indicate whether the string is just a
normal string or base64 encoded binary data.

On Mon, Feb 23, 2015 at 1:54 PM, Tim Hockin <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4514#discussion_r25190058>
> :
>
> > @@ -1309,3 +1317,27 @@ type ResourceQuotaList struct {
> >  	// Items is a list of ResourceQuota objects
> >  	Items []ResourceQuota `json:"items"`
> >  }
> > +
> > +// Secret holds secret data of a certain type
> > +type Secret struct {
> > +	TypeMeta   `json:",inline"`
> > +	ObjectMeta `json:"metadata,omitempty"`
> > +
> > +	Data map[string][]byte `json:"data,omitempty"`
>
> Also, @bgrant0607 <https://github.com/bgrant0607> This uses a map - more
> fodder for maps being the OBVIOUS api for things like environment variables.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4514/files#r25190058>
> .
>